### PR TITLE
vdk-control-service: potential resource leak fixes

### PR DESCRIPTION
**Pull request by dakodakov**
Why?
Fix potential resource leak fixes uncovered by coverity scaner

What?
Switch to try-with-resource blocks

What type of change are you making?
Bug fix

Signed-off-by: Dako Dakov ddakov@vmware.com

### CODE REVIEW

**Review by murphp15 — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java [hunk 1]
```diff
@@ -100,14 +100,17 @@ public static File unzipDataJob(Resource resource, File tempDir, String jobName)
     File jobZipFile = new File(tempDir, "job_zip.zip");
     File jobTempDir = new File(tempDir, "job_dir");
     org.apache.commons.io.FileUtils.copyInputStreamToFile(resource.getInputStream(), jobZipFile);
-    new ZipFile(jobZipFile).extractAll(jobTempDir.getAbsolutePath());
+    try (ZipFile theZipFile = new ZipFile(jobZipFile)) {
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java:103]: so ZipFile close implementation would handle openInputStreams and there are none, so no actual resources leak is patched. However it's a good style to close Closable objects upon usage, due to in the future implementation may change.

**Review by ivakoleva — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java (hunk 1 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java:103]: Hey @ivakoleva , 
Could you elaborate please? The theZipeFile object is declared in a try-with-resource block and is going to be automatically closed.

Comment (ivakoleva) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java:103]: the PR is named "potential resource leak fixes", I look to understand are there any leaks or not? This helps classify the old release severity, new release adoption and prioritization of adoption/fix. 
The ZipFile implements Closable, so it should be closed. In terms of severity, there was no actual resource leak.

Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java:103]: ZipFile also implements AutoClosable, which allows the class to be used in try-with-resource block and after the block is finished, the JVM will automatically close it. 
This change is to address a Coverity scan warning for a _potential_ resource leak which could occur if the usage of the ZipFile object causes an exception which was previously not handled in this method.

Comment (ivakoleva) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java:103]: The automated tools could hint you where to look;

**Pull request merged**